packages/simplr-forms/src/abstractions/base-field.ts

```diff
@@ -19,6 +28,29 @@ export abstract class BaseField<TProps extends FieldProps, TState extends BaseFi
         return this.props.value;
     }
 
+    protected get IsControlled(): boolean {
+        return this.props.value != null;
```

Comment:
Good catch. `props.value` should be `undefined` if it's empty and the check here should be `this.props.value !== undefined`. Will fix.